modules/python/python-dest.c

```diff
@@ -199,6 +202,26 @@ _py_invoke_deinit(PythonDestDriver *self)
   _dd_py_invoke_void_method_by_name(self, "deinit");
 }
 
+static PyObject *
+_py_dest_get_seqnum(PyObject *s)
+{
+  PythonDestDriver *self = PyLong_AsVoidPtr(s);
+  return py_as_int(self->super.seq_num);
+}
+
+static void
+register_sequence_number(PythonDestDriver *self)
+{
+  static PyMethodDef getSeqNumMethod = {"getSeqNum", (PyCFunction)_py_dest_get_seqnum, METH_VARARGS, "Read sequence number"};
+  PyObject *dest_driver = PyLong_FromVoidPtr(self);
+  PyObject *func = PyCFunction_New(&getSeqNumMethod, dest_driver);
+  Py_DECREF(dest_driver);
+  self->py.seqnum_method = slng_PyMethod_New(func, self->py.class);
+  Py_DECREF(func);
+  PyObject_SetAttrString(self->py.class, "getSeqNum", self->py.seqnum_method);
+  Py_DECREF(self->py.seqnum_method);
+}
```

Comment:
it's much easier to register methods through the use of the tp_methods member of the PyType that you register. (see python-logmsg for an example). Is there a specific reason why you are doing it this way?

Reply:
I think I now understand, the object is instantiated by a Python class that does not derive from any superclass, this way we can't publish methods from our C code.

I think that's a design issue in the interface that we should - eventually - rectify. Backward compatibility could be an issue, but I think it would be a good tradeof that if you want the new fields (e.g. template_options and getSeqNum), you need to derive your class from syslogng.DestinationDriver or something like that. The java interface also uses base classes and I think that's a good solution.